File: common/src/main/java/com/ssblur/scriptor/helpers/targetable/SpellbookTargetable.java
package com.ssblur.scriptor.helpers.targetable;

import net.minecraft.world.entity.player.Player;
import net.minecraft.world.item.ItemStack;

public class SpellbookTargetable extends ItemTargetable{
  int slot;
  public SpellbookTargetable(ItemStack itemStack, Player entity, int slot) {
    super(itemStack, entity);
    this.slot = slot;
  }

  public Player getPlayer() {
    if(targetEntity instanceof Player player)
      return player;
    return null;
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/helpers/targetable/SpellbookTargetable.kt
package com.ssblur.scriptor.helpers.targetable

import net.minecraft.world.entity.player.Player
import net.minecraft.world.item.ItemStack

class SpellbookTargetable(itemStack: ItemStack?, entity: Player?, var slot: Int) : ItemTargetable(
    itemStack!!, entity
) {
    val player: Player?
        get() {
            if (targetEntity is Player) return targetEntity as Player
            return null
        }
}
